### PR TITLE
Compute balanced accuracy metric

### LINKED ISSUE #27: Add column `balanced_accuracy` to the results table of `morphoclass performance-table`
## Context
- When we run `morphoclass performance-table`, we only report `accuracy` and `f1_score` with various averaging methods (`macro`, `micro`, `weighted`):
Screenshot from 2022-04-20 15-51-45
- But since our dataset it would also be useful to also compute the [`balanaced_accuracy`](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.balanced_accuracy_score.html#sklearn.metrics.balanced_accuracy_score), which is basically an `accuracy` with `weighted` averaging.

## Accuracy
- [ ] Add to the HTML reports of `morphoclass performance-table` a new column  `balanced_accuracy`.

### PR DESCRIPTION
**Pull request by FrancescoCasalegno**
Fixes #27.

## Description

1. The table created by ` morphoclass performance-table` now contains a `"balanced_accuracy"` column.
2. The table is now written both to the HTML report _and_ to a CSV file (for further manipulation, in view of the paper).

## TODO

- [ ] All metrics should be computed in the `make_performance_table()` function (even though they are already written in the checkpoint at training time).

## How to test?

Please provide here instructions on how to test the changes introduced by this PR.
(if some changes cannot be tested by automated tests)

## Checklist

- [x] This PR refers to an issue from the [issue tracker](https://github.com/BlueBrain/morphoclass/issues).
  (if it is not the case, please create an issue first).
- [x] All CI tests pass.

### CODE REVIEW

**Comment by FrancescoCasalegno** [comment]
## Update 2022-05-03

Except for the only metric `"balanced_accuracy"`, all other metrics (`"accuracy"`, `"f1_weighted"`, `"f1_micro"`, `"f1_macro"`) returned by the function `make_report_row()` are currently read from the loaded checkpoint, i.e. the object `data`.

However we get the very same results (both `mean` and `std`) if we compute those metrics in `make_report_row()`. This would also make more sense, since the computation of evaluation metrics should be done in the evaluation stages, not when saving the training checkpoint!

However, this can be handled in a separate PR.
